cwc-infrastructure/usr/share/clearwater/infrastructure/scripts/clearwater-decommission.sh

```
@@ -0,0 +1,218 @@
+#!/usr/bin/env bash
+
+###############################################################################
+# Copyright (C) 2016 Metaswitch Networks, Inc.
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later 
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or 
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for 
+# more details (http://www.gnu.org/licenses)
+###############################################################################
+# Allowing logging and checking that the user is root and set error stops
+set -e
+if [ "$(whoami)" != "root" ]
+then
+    echo "Error: must be run as root."
+    exit 1
+fi
+. /usr/share/clearwater/infrastructure/scripts/logging.bash 'decommission.log' 'INFO'
+###############################################################################
+
+
+# This function uses the clearwater-version command to find a nodes name 
+# (sprout, ralf or homestead). This is written with the expectation that
+# a single node will only contain one of the three types and not a combination.
+function getNodeName()
```

Can we return an error code if this fails. E.g. 'return 1' if NODE_NAME is unset at the end.